Status should be not ok if flush a table not existed

**Describe the bug**
```
    def test_flush_table_not_existed(self, connect, table):
        '''
        target: test delete vector, params table_name not existed
        method: add vector and delete
        expected: status not ok
        '''
        table_new = gen_unique_str()
        status = connect.flush([table_new])
>       assert not status.OK()
E       AssertionError: assert not True
E        +  where True = <bound method Status.OK of Status(code=0, message='OK')>()
E        +    where <bound method Status.OK of Status(code=0, message='OK')> = Status(code=0, message='OK').OK

```

**Steps/Code to reproduce behavior**
**Expected behavior**

**Environment details**
branch: crud

**Screenshots**

**Additional context**

--- a/core/src/db/DBImpl.cpp
+++ b/core/src/db/DBImpl.cpp
@@ -537,9 +537,18 @@ DBImpl::Flush(const std::string& table_id) {
         return SHUTDOWN_ERROR;
     }
 
+    bool has_table;
+    auto status = HasTable(table_id, has_table);
+    if (!has_table) {
+        ENGINE_LOG_ERROR << "Table to flush does not exist: " << table_id;
+        return Status(DB_NOT_FOUND, "Table to flush does not exist");
+    }
+    if (!status.ok()) {
+        return Status(DB_ERROR, status.message());
+    }
+
     ENGINE_LOG_DEBUG << "Flushing table: " << table_id;
 
-    Status status;
     if (wal_enable_ && wal_mgr_ != nullptr) {
         auto lsn = wal_mgr_->Flush(table_id);
         if (lsn != 0) {

--- a/core/unittest/db/test_db.cpp
+++ b/core/unittest/db/test_db.cpp
@@ -772,3 +772,8 @@ TEST_F(DBTestWAL, DB_TEST) {
     stat = db_->DropTable(table_info.table_id_, dates);
     ASSERT_TRUE(stat.ok());
 }
+
+TEST_F(DBTest2, flush_non_existing_table) {
+    auto status = db_->Flush("non_existing_table");
+    ASSERT_FALSE(status.ok());
+}

